merkledag/merkledag.go

```diff
@@ -269,3 +313,24 @@ func (np *nodePromise) Get(ctx context.Context) (*Node, error) {
 	}
 	return np.cache, nil
 }
+
+// EnumerateChildren will walk the dag below the given root node and add all
+// unseen children to the passed in set.
+// TODO: parallelize to avoid disk latency perf hits?
```

Comment:
if and when it becomes a bottleneck for some operation, we can introduce a bunch of additional complexity to add concurrency here. In my tests on a datastore of around 2GB, it took less than 8 seconds to run over the entire set of blocks.